pika/channel.py

```diff
@@ -1156,6 +1159,8 @@ def _on_getempty(self, method_frame):
 
         """
         LOGGER.debug('Received Basic.GetEmpty: %r', method_frame)
+        if self._on_getok_callback is not None:
```

Comment:
@dschep: there is no need to check `if self._on_getok_callback is not None` before setting `self._on_getok_callback = None`. The check is redundant. Just setting `self._on_getok_callback = None` would suffice.